contributors/design-proposals/auth/bound-service-account-tokens.md

```diff
@@ -213,6 +213,67 @@ The token payload will be:
 }
 ```
 
+### Add Service Account signing key retrieval API
+
+A non-resource API will be added to the apiserver that will expose an [OIDC
+discovery](https://openid.net/specs/openid-connect-discovery-1_0.html)
+compatible configuration. This will allow clients to retrieve a list of
+currently valid signing keys as an array of JWKs. For example, if the service
+account issuer is configured with the name `https://dev.cluster.internal`, the
+apiserver could expose the following configuration:
+
+```
+> GET /.well-known/openid-configuration
+{
+  "issuer": "https://dev.cluster.internal",
+  "jwks_uri": "https://dev.cluster.internal/serviceaccountkeys/v1",
+  "authorization_endpoint": "urn:kubernetes:programmatic_authorization",
```

Comment:
is this compatible with https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata:
```
authorization_endpoint
    REQUIRED. URL of the OP's OAuth 2.0 Authorization Endpoint
```
that specifies a URL, not a URI (which would allow a URN)

Reply:
Does `authorization_endpoint` even make sense?  There is no way to actually do an OAuth flow with the API master.